pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -161,10 +165,24 @@ func validateSystemRequirements(mountUtil mount.Interface) (features, error) {
 // Takes the absolute name of the specified containers.
 // Empty container name disables use of the specified container.
 func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.Interface, nodeConfig NodeConfig) (ContainerManager, error) {
+	// Check if Cgroup-root actually exists on the node
+	if nodeConfig.CgroupsPerQOS {
+		if _, err := os.Stat(nodeConfig.CgroupRoot); err != nil {
+			return &containerManagerImpl{}, fmt.Errorf("invalid configuration: CgroupRoot doesn't exist : %v", err)
+		}
+	}
+	if nodeConfig.CgroupsPerQOS && nodeConfig.CgroupRoot == "" {
```

Comment:
this check should come before the check to see if the file directory exists above